### ISSUE TITLE
undefined method 'op_asgn_type?' error with syntactically valid code

### ISSUE THREAD

**Issue by CampNowhere** [rubocop/rubocop#13458]
Rubocop 0.93.1 (using Parser 3.1.2.1, rubocop-ast 1.23.0, running on ruby 3.1.4 x86_64-linux)

I get this error when running this block of code, which is syntactically correct:

```ruby
# frozen_string_literal: true

require 'spec_helper'

RSpec.describe MigrationFailureNotice do
  describe 'end to end test for rendering a migration failure notice' do
    let(:fixture_root) { 'v4' }

    let(:result) { MultiDbResult.from_directory(file_fixture("migration-testing/#{fixture_root}")) }

    let(:expected_migration_failure_notice) do
      file_fixture("migration-testing/#{fixture_root}/expected-migration-failure-notice.txt")
    end

    subject { described_class.new(result).render }

    before do
      override_env_from_fixture('migration-testing/environment.json')
    end

    # The expectation for this spec lives in `expected_comment_file`
    # It can be re-recorded with: `RECAPTURE_END_TO_END_RESULTS=1 bundle exec rspec spec`
    it 'renders the comment for fixtures' do
      if ENV['RECAPTURE_END_TO_END_RESULTS']
        File.open(expected_migration_failure_notice, 'wb+') do |io|
          io << subject
        end
      end

      expect(subject).to eq(File.read(expected_migration_failure_notice))
    end
  end
end
```

If it's of any help as well, the main message was `An error occurred while Layout/BlockAlignment cop was inspecting [file]`

Finally, this code parses just fine using regular `ast` with `Parser::Current`.

**Comment by dvandersluis** [comment]
You’re using a really old version of rubocop, can you please try updating it and see if the issue persists?

**Comment by CampNowhere** [comment]
The issue persists if I update to the max version supported by Ruby 3.1.4.

**Comment by dvandersluis** [comment]
The current version of rubocop is [supported by Ruby 3.1](https://docs.rubocop.org/rubocop/1.68/compatibility.html#support-matrix). Is that what you're using? Your code doesn't register any offenses for me on v1.68.

If you're getting offenses still, please share your rubocop configuration.

**Comment by vlad-pisanov** [comment]
`undefined method op_asgn_type?` in layout cops sounds a lot like https://github.com/rubocop/rubocop/issues/10258

**Comment by dvandersluis** [comment]
Good find @vlad-pisanov!

Strangely, I still cannot replicate even when the same gem versions (`0.93.1 (using Parser 3.1.2.1, rubocop-ast 1.23.0, running on ruby 3.1.4 arm64-darwin22)`).

But in any case, @CampNowhere it seems like 1.22.3 or newer (https://github.com/rubocop/rubocop/issues/10258#issuecomment-974615729) should fix this for you, or you can disable the `Layout/BlockAlignment` cop until you are able to upgrade.
